server/src/main/java/org/elasticsearch/cluster/metadata/DataStream.java

```diff
@@ -265,24 +268,24 @@ public IndexMode getIndexMode() {
      *
      * @param writeIndex            new write index
      * @param generation            new generation
-     * @param indexModeFromTemplate the index mode as is defined in the template that created this data stream
+     * @param isTsdbTemplate the index mode as is defined in the template that created this data stream
      *
      * @return new {@code DataStream} instance with the rollover operation applied
      */
-    public DataStream rollover(Index writeIndex, long generation, IndexMode indexModeFromTemplate) {
+    public DataStream rollover(Index writeIndex, long generation, boolean isTsdbTemplate) {
```

Comment:
This is internal, so it is probably ok to keep it a boolean here, but I would still prefer not to use the abbreviation Tsdb. I think we are calling it TimeSeries everywhere else in the code.